pallets/collator-allowlist/src/lib.rs

```diff
@@ -171,6 +171,7 @@ impl<T: Config> ValidatorRegistration<T::ValidatorId> for Pallet<T> {
 	/// True iff
 	///   - the validator id is present in the allowlist and
 	///   - the validator id is registered in the underlying validator registration center
+	#[cfg(not(test))]
 	fn is_registered(id: &T::ValidatorId) -> bool {
 		let contains_key = if cfg!(feature = "runtime-benchmarks") {
```

Comment:
I think you can also do:
```suggestion
		let contains_key = if cfg!(all(test, feature = "runtime-benchmarks")) {
```

Reply:
Nah, this isn't what I want. We are building with `runtime-benchmakrs` during tests. But the tests actually need the logic to behave like it does when not using this flag.

Reply:
Not saying that there might be a nicer, cleaner solution ^^